From b0d9a6accad148db070d27fffe2131667a7f04e8 Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Mon, 24 Aug 2020 17:12:12 +0200
Subject: [PATCH] Customize ST GTK demo-launcher for PHYTEC board

Add PHYTEC logo and phyCORE-STM32MP image.
Add SOM name and Baseboard name.
Use Phytec green color for background.

Signed-off-by: Christophe Parant <c.parant@phytec.fr>
---
 demo_launcher.py | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/demo_launcher.py b/demo_launcher.py
index cc6e8a3..b3e647f 100755
--- a/demo_launcher.py
+++ b/demo_launcher.py
@@ -163,7 +163,7 @@ def _load_image_eventBox(parent, filename, label_text1, label_text2, scale_w, sc
 
     return eventBox
 
-def _load_image_Box(parent, mp1filename, infofilename, label_text, scale_w, scale_h):
+def _load_image_Box(parent, mp1filename, infofilename, label_text1, label_text2, scale_w, scale_h):
     box = Gtk.VBox(homogeneous=False, spacing=0)
     pixbuf = GdkPixbuf.Pixbuf.new_from_file_at_scale(
             filename=mp1filename,
@@ -175,10 +175,11 @@ def _load_image_Box(parent, mp1filename, infofilename, label_text, scale_w, scal
     # Create a label for the button
     label0 = Gtk.Label() #for padding
     label1 = Gtk.Label()
-    label1.set_markup("<span font='14' color='#FFFFFFFF'><b>%s</b></span>\n"
-                      "<span font='10' color='#FFFFFFFF'>Dual Arm&#174; Cortex&#174;-A7</span>\n"
-                      "<span font='10' color='#FFFFFFFF'>+</span>\n"
-                      "<span font='10' color='#FFFFFFFF'>Copro Arm&#174; Cortex&#174;-M4</span>\n" % label_text)
+    label1.set_markup("<span font='14' color='#00000000'><b>%s</b></span>\n"
+		      "<span font='12' color='#00000000'>%s</span>\n"
+                      "<span font='10' color='#00000000'>Dual Arm&#174; Cortex&#174;-A7</span>\n"
+                      "<span font='10' color='#00000000'>+</span>\n"
+                      "<span font='10' color='#00000000'>Copro Arm&#174; Cortex&#174;-M4</span>\n" % (label_text1, label_text2))
     label1.set_justify(Gtk.Justification.CENTER)
     label1.set_line_wrap(True)
 
@@ -194,7 +195,7 @@ def _load_image_Box(parent, mp1filename, infofilename, label_text, scale_w, scal
     eventBox.connect("button_press_event", parent.highlight_eventBox)
 
     label3 = Gtk.Label()
-    label3.set_markup("<span font='10' color='#FFFFFFFF'><b>Python GTK launcher</b></span>\n")
+    label3.set_markup("<span font='10' color='#00000000'><b>Python GTK launcher</b></span>\n")
     label3.set_justify(Gtk.Justification.CENTER)
     label3.set_line_wrap(True)
 
@@ -462,7 +463,7 @@ def gtk_style():
     background-color: rgba (0%, 0%, 0%, 0.1);
 }
 #logo_bg {
-    background-color: rgba (31%, 32%, 31%, 1.0);
+    background-color: rgba (82%, 88%, 89%, 1.0);
 }
 #backed_bg {
     background-color: rgba (31%, 32%, 31%, 0.8);
@@ -502,9 +503,11 @@ class MainUIWindow(Gtk.Window):
                 self.screen_height = self.get_screen().get_height()
 
         if self.screen_width == 720:
-            self.board_name = "Evaluation board"
+            self.board_name = "phyBOARD-Sargas"
         else:
-            self.board_name = "Discovery kit"
+            self.board_name = "phyBOARD-Sargas"
+
+        self.som_name="phyCORE-STM32MP1"
 
         if min(self.screen_width, self.screen_height) >= 720:
             self.icon_size = ICON_SIZE_BIG
@@ -594,7 +597,7 @@ class MainUIWindow(Gtk.Window):
         self.icon_grid.set_row_spacing(20)
 
         # STM32MP1 Logo and info area
-        self.logo_info_area = _load_image_Box(self, "%s/pictures/ST11249_Module_STM32MP1_alpha.png" % DEMO_PATH, "%s/pictures/ST13340_Info_white.png" % DEMO_PATH, self.board_name, -1, 160)
+        self.logo_info_area = _load_image_Box(self, "%s/pictures/phyCORE-STM32MP1.png" % DEMO_PATH, "%s/pictures/ST13340_Info_white.png" % DEMO_PATH, self.som_name, self.board_name, -1, 160)
         self.logo_info_area.set_name("logo_bg")
         self.icon_grid.attach(self.logo_info_area, 3, 0, 1, 2)
 
-- 
2.7.4

